Automattic/jetpack
# Deprecation notices: Add getRedirectUrl to ensure correct URL creation

## Proposed changes:

* This PR adds the `getRedirectUrl` function to aid in support document URL creation for deprecation notices. The url  passed to Jetpack Settings and Dashboard notices (passed via the `DeprecationNotice` component) was missing `getRedirectUrl` to generate the support URL.  This was spotted as missing whilst writing up documentation on the notices here: https://github.com/Automattic/jetpack/pull/40054
* This existed in the original [Google Analytics deprecation code](https://github.com/Automattic/jetpack/pull/38701/files#diff-d45e0ec2af07da4512ee3eb49625d9a42248ae1d11f5d537fbbd131899441bb4) so is just a case of adding it back in.

### Other information:


## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* To test the `getRedirectUrl` addition, best testing for that would be to add in `jetpack-support` for the `url` property when testing locally, in the demo notice. Notice testing is covered here: https://github.com/Automattic/jetpack/pull/39567. It is the notice showing on the Jetpack Settings / Dashboard page that we'd be testing in this case, so we only need to build the Jetpack plugin if testing locally.
